File: 14-longest-common-prefix/longest-common-prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        strs.sort()
        print(strs)
        res=0
        hashmap=defaultdict(int)
        print(hashmap)
        min_lenght=min([len(strings) for strings in strs])
        print(min_lenght)
        for i in range(min_lenght):
            for strings in strs:
                hashmap[strings[i]]+=1
            print(hashmap,len(hashmap))
            if len(hashmap) ==1:
                res+=1
            else:
                break
            hashmap.clear()
        print(res)
        if res==0:
            return ""
        return strs[0][:res]
        

File: 14-longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        int minum_length=strs[0].length();
        int minimum_prefix_length=0;
        int k=0;
        //nadiiiiiiiiii
        if (strs == null || strs.length == 0) {
            return "";
        }
        for(int i=1;i<strs.length;i++){
            if(strs[i].length()<minum_length){
                minum_length=strs[i].length();
            }
        }
        while(k<minum_length){
            for(int j=0;j<strs.length;j++){
                if(strs[j].charAt(k)!=strs[0].charAt(k)){
                    return strs[0].substring(0,k);
                }
            }
            k++;  
        }
        return strs[0].substring(0, minum_length);
    }
    
}